frame/proxy/src/lib.rs

```diff
@@ -597,6 +564,60 @@ impl<T: Trait> Module<T> {
 		T::AccountId::decode(&mut &entropy[..]).unwrap_or_default()
 	}
 
+	pub fn add_proxy_delegate(
+		delegator: &T::AccountId,
+		delegatee: T::AccountId,
+		proxy_type: T::ProxyType,
+		delay: T::BlockNumber,
+	) -> DispatchResult {
+		Proxies::<T>::try_mutate(delegator, |(ref mut proxies, ref mut deposit)| {
+			ensure!(proxies.len() < T::MaxProxies::get() as usize, Error::<T>::TooMany);
+			let proxy_def = ProxyDefinition { delegate: delegatee, proxy_type, delay };
+			let i = proxies.binary_search(&proxy_def).err().ok_or(Error::<T>::Duplicate)?;
+			proxies.insert(i, proxy_def);
+			let new_deposit =Self::deposit(proxies.len() as u32);
+			if new_deposit > *deposit {
```

Comment:
Why not take the depositing as parameter of the function? A closure or something similar. In this way you could ignore the deposit in your pallet if you like?

Reply:
User will still be able to manage proxies via this pallet directly so I was worried about all the edge cases.

For add proxy, we can just not taking a deposit, but for remove proxy, we should refund the deposit, but because add proxy may or may not take deposit, the code is then start getting complicated...

I still want to keep this change as minimal as possible.